File: TraceEtw/TraceEtwTests.Windows/unittest1.cpp
#include "pch.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace Diagnostics;
using namespace Platform;
using namespace Platform::Collections;
using namespace Windows::Storage;

TEST_CLASS(UnitTest1)
{
public:

    TEST_METHOD(CX_Windows_TestBasic)
    {
        StorageFile^ file;

        {
            auto listener = ref new InProcEventListener(
                ApplicationData::Current->LocalFolder,
                L"CX_Windows_TestBasic.etl",
                ref new Vector < Guid > { MMaitre_TraceEtw }
            );

            file = Await(listener->GetLogFileAsync());

            Logger::WriteMessage(file->Path->Data());
            Logger::WriteMessage("\n");

            Assert::IsTrue(EtwLogger.IsEnabled());

            EtwLogger.Trace("1");
            EtwLogger.Error("2");
            EtwLogger.OpStart(nullptr);
            EtwLogger.OpStop(nullptr, E_FAIL);
            EtwLogger.TraceVarArgsAnsi("Value: %i", 0);
            EtwLogger.TraceVarArgsUnicode(L"Record: %s", L"foo");
            EtwLogger.Marker();
            EtwLogger.MarkerOpStart();
            EtwLogger.MarkerOpStop();
            EtwLogger.MarkerOp2Start();
            EtwLogger.MarkerOp2Stop(&GUID_NULL);
            EtwLogger.ManyArgs(true, 0, 0, 0, 0, 0, 0, 0, 0, 0.f, 0., &GUID_NULL, nullptr, "", L"");
        }

        auto props = Await(file->GetBasicPropertiesAsync());
        Assert::IsTrue(props->Size > 0);

        auto folder = Await(KnownFolders::SavedPictures->CreateFolderAsync("Tests", CreationCollisionOption::OpenIfExists));
        Await(file->MoveAsync(folder, "CX_Windows_TestBasic.etl", NameCollisionOption::ReplaceExisting));

        Logger::WriteMessage(file->Path->Data());
    }

};

File: TraceEtw/EventListenerTests.Windows/UnitTest1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;
using Diagnostics;
using Windows.Storage;
using System.Diagnostics.Tracing;
using System.Threading.Tasks;

namespace EventListenerTests.Windows
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public async Task CS_Windows_TestBasic()
        {
            StorageFile file;

            using (var listener = new InProcEventListener(
                ApplicationData.Current.LocalFolder, 
                "CS_Windows_TestBasic.etl", 
                new Guid[] { Log.Events.Guid }
                ))
            {
                file = await listener.GetLogFileAsync();

                Logger.LogMessage("Log file: {0}", file.Path);

                Assert.IsTrue(Log.Events.IsEnabled());

                Log.Events.ProcessImageStart();
                Log.Events.Message("A message");
                Log.Events.ProcessImageStop();
            }

            var props = await file.GetBasicPropertiesAsync();
            Assert.IsTrue(props.Size > 0);

            var folder = await KnownFolders.SavedPictures.CreateFolderAsync("Tests", CreationCollisionOption.OpenIfExists);
            await file.MoveAsync(folder, "CS_Windows_TestBasic.etl", NameCollisionOption.ReplaceExisting);

            Logger.LogMessage("Log file moved to: {0}", file.Path);
        }
    }
}
